File: src/main/java/com/homechange/api/rest/dto/user/UserResponseDTO.java
package com.homechange.api.rest.dto.user;

import com.homechange.api.model.User;

import java.util.Date;

/**
 * Created by Nemanja on 5/14/17.
 *
 * DTO Used in responses, contains user id
 */
public class UserResponseDTO extends UserDTO {

	private Long userId;

	public UserResponseDTO() {
		super();
	}

	/**
	 * Constructor to be used for mapping repository object to response
	 * @param user User
	 */
	public UserResponseDTO(User user) {
		super(user.getUsername(), user.getFirstName(), user.getLastName(), user.getEmail(), user.getCountry(),
				user.getCity(), user.getAddress(), user.getDateOfBirth(), user.getPhoneNumber());
		this.userId = user.getId();
	}

	public Long getUserId() {
		return userId;
	}

	public void setUserId(Long userId) {
		this.userId = userId;
	}
}

File: src/main/java/com/homechange/api/rest/dto/user/UserResponseDTO.kt
package com.homechange.api.rest.dto.user

import com.homechange.api.model.User

/**
 * Created by Nemanja on 5/14/17.

 * DTO Used in responses, contains user id
 */
class UserResponseDTO : UserDTO {

    var userId: Long? = null

    constructor() : super() {}

    /**
     * Constructor to be used for mapping repository object to response
     * @param user User
     */
    constructor(user: User?) : super(user?.username, user?.firstName, user?.lastName, user?.email, user?.country,
            user?.city, user?.address, user?.dateOfBirth, user?.phoneNumber) {
        this.userId = user?.id
    }
}
